methods.py

```diff
@@ -69,6 +69,7 @@ def add_source_files_scu(self, sources, files, allow_gen=False):
 
         # Add all the gen.cpp files in the SCU directory
         add_source_files_orig(self, sources, subdir + ".scu/scu_*.gen.cpp", True)
+        self.Append(CPPDEFINES=["SCU_BUILD"])
```

Comment:
I would personally tend towards `SCU_BUILD_ENABLED` as we tend to have e.g. `ANDROID_ENABLED`, `DEV_ENABLED` etc as a convention (maybe we might want to use e.g. `SCU_BUILD` in a macro in future or something).

Also just to check, my python-fu isn't strong, is there any possibility that we could end up defining this more than once in the same environment? (I.e. could the compiler command line end up having lots of similar defines?)

Reply:
I pushed an update with your suggested insertion position. This seems to work as well.